str.go

```diff
@@ -13,14 +19,17 @@ import (
 // STR should be final
 type SignedTreeRoot struct {
 	tree        *MerkleTree
-	counter     int64
-	prevCounter int64
+	counter     uint64
```

Comment:
Hmm, does it make sense? I mean now `epoch` is `uint64` type, and `prevEpoch` is just `epoch`- 1.